parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReader.java

```diff
@@ -76,18 +87,12 @@ public void initFromPage(int valuesCount, ByteBufferInputStream stream)
       throw new ParquetDecodingException(errorMessage);
     }
 
-    // Allocate buffer for all of the byte stream data.
+    // Eagerly read and decode the data. This allows returning stable
+    // Binary views into the internal decode buffer for FIXED_LEN_BYTE_ARRAY.
     final int totalSizeInBytes = stream.available();
-    byteStreamData = new byte[totalSizeInBytes];
-
-    // Eagerly read the data for each stream.
-    final int numRead = stream.read(byteStreamData, 0, totalSizeInBytes);
-    if (numRead != totalSizeInBytes) {
-      String errorMessage = String.format(
-          "Failed to read requested number of bytes. Expected: %d. Read %d.", totalSizeInBytes, numRead);
-      throw new ParquetDecodingException(errorMessage);
```

Comment:
It seems that this check has been changed into assert somewhere else. I'd suggest to throw if unexpected padding or short read are found.

Reply:
Are there situations where `stream.slice` below would return something smaller than what `stream.available` returned?

Reply:
No, but we do have seen cases where `numRead < totalSizeInBytes` so there is a clarification here: https://github.com/apache/parquet-format/commit/230711fbfd8d3399cce935a4f39d1be7b6ad5ad5

Reply:
Well, I don't understand:
1. The `ByteBufferInputStream` declares that it has `totalSizeInBytes` bytes available
2. You ask the `ByteBufferInputStream` to read  `totalSizeInBytes` bytes
3. You end up with less than  `totalSizeInBytes` bytes read??

It seems unrelated with padding. It's just the physical size of the stream.

However, what could happen is that `totalSizeInBytes` is not a multiple of `elementSizeInBytes` for some reason, and currently this would hit the assertion at the end of `decodeData`. I could instead add a proper exception for this.

Reply:
(I understand that it might theoretically happen if `ByteBufferInputStream` is being read from by another thread, but that sounds like an API violation, doesn't it?)

Reply:
It seems the check was originally useful, but it became trivially true after this change:
https://github.com/apache/parquet-mr/commit/bf38e0fe549460936b310764a762749cb351ae6c#diff-215039b16c580f61a577c71dd878b2acf390d510229b9d190d8738fac43b1a6eR80-R86

Reply:
Yes, you're right.